src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -125,6 +125,8 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 
 for scope in ['vm', 'vmss']:
     register_cli_argument(scope, 'no_auto_upgrade', action='store_true', help='by doing this, extension system will not pick the highest minor version for the specified version number, and will not auto update to the latest build/revision number on any scale set updates in future.')
+    register_extra_cli_argument('{} create'.format(scope), 'generate_ssh_keys', action='store_true', help='Generate SSH public and private key files if missing')
```

Comment:
Since VM/VMSS create are custom commands, it would make more sense to simply add them to the respective command signatures. "register_extra_cli_argument" is more for adding things to SDK methods.